plumgrid_playbooks/roles/plumgrid_director/tasks/main.yml

```diff
@@ -14,26 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Extract License from file
 - name: Extract License
   set_fact:
     pg_license: "{{ lookup('file', pg_license_path) }}"
 
-# Modify template files
-- name: Setup Keepalived Config
+- name: Setup Keepalived and Nginx config
   template:
-    src: keepalived.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/etc/keepalived.conf
-
-- name: Setup nginx Config
-  template:
-    src: default.conf
-    dest: /var/lib/libvirt/filesystems/plumgrid-data/conf/pg/nginx.conf
-
-- name: Restart PLUMgrid service
-  service:
-    name: plumgrid
-    state: restarted
+    src: "{{ item.src }}"
+    dest: "{{ plumgrid_data_dir }}/{{ item.dest }}"
+  with_items:
```

Comment:
@arslanqadeer are you testing it from the scratch? if not, please test it from the scratch. Notify only work if there is any change, before you were restarting the service without checking that it require or not. Plus this is wrong format I think.

```
name: Restart PLUMgrid
service: name=plumgrid state=restarted
notify: Restart PLUMgrid Service
```